Makefile

```diff
@@ -44,13 +50,19 @@ docker-%: ## Builds the specified target defined in the Dockerfile using the doc
 	@$(MAKE) target-$* TARGET_ARGS="$(TARGET_ARGS)"
 
 .PHONY: $(TARGETS) $(NONFREE_TARGETS)
-$(TARGETS) $(NONFREE_TARGETS):
-	@$(MAKE) docker-$@ TARGET_ARGS="--tag=$(REGISTRY)/$(USERNAME)/$@:$(TAG) --push=$(PUSH)"
+$(TARGETS) $(NONFREE_TARGETS): $(ARTIFACTS)/bldr
+	@$(MAKE) docker-$@ \
+		TARGET_ARGS="--tag=$(REGISTRY)/$(USERNAME)/$@:$(shell $(ARTIFACTS)/bldr eval --target $@ --build-arg TAG=$(TAG) '{{.VERSION}}' 2>/dev/null) --push=$(PUSH)"
 
 .PHONY: deps.png
-deps.png:
-	bldr graph | dot -Tpng > deps.png
+deps.png: $(ARTIFACTS)/bldr
+	$(ARTIFACTS)/bldr graph | dot -Tpng > deps.png
 
 .PHONY: conformance
 conformance: ## Performs policy checks against the commit and source code.
 	docker run --rm -it -v $(PWD):/src -w /src ghcr.io/siderolabs/conform:latest enforce
+
+$(ARTIFACTS)/bldr:
+	mkdir -p $(ARTIFACTS)
+	curl -L https://github.com/siderolabs/bldr/releases/download/$(BLDR_RELEASE)/bldr-linux-amd64 -o $(ARTIFACTS)/bldr
```

Comment:
should we support both arch to be generic

Reply:
sure, fixed!